.gitlab-ci.yml

```diff
@@ -34,7 +34,7 @@ variables:
   RECOVER_CONTROL_PLANE_TEST: "false"
   RECOVER_CONTROL_PLANE_TEST_GROUPS: "etcd[2:],kube_control_plane[1:]"
   TERRAFORM_VERSION: 1.0.8
-  ANSIBLE_MAJOR_VERSION: "2.10"
+  ANSIBLE_MAJOR_VERSION: "2.11"
```

Comment:
The following error is surely due to this change:
```
TASK [container-engine/nerdctl : download_file | Copy file back to ansible host file cache] ***
task path: /builds/kargo-ci/kubernetes-sigs-kubespray/roles/download/tasks/download_file.yml:102
Monday 06 June 2022  22:14:53 +0000 (0:00:01.358)       0:03:12.206 *********** 
redirecting (type: action) ansible.builtin.synchronize to ansible.posix.synchronize
redirecting (type: action) ansible.builtin.synchronize to ansible.posix.synchronize
redirecting (type: action) ansible.builtin.synchronize to ansible.posix.synchronize
redirecting (type: action) ansible.builtin.synchronize to ansible.posix.synchronize
fatal: [instance-1]: FAILED! => {"changed": false, "cmd": "/usr/bin/rsync --delay-updates -F --compress --archive --rsh='/usr/bin/ssh -S none -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null' --rsync-path='sudo -u root rsync' --out-format='<<CHANGED>>%i %n%L' kubespray@instance-1:/tmp/kubespray_cache/nerdctl-0.19.0-linux-amd64.tar.gz /tmp/kubespray_cache/nerdctl-0.19.0-linux-amd64.tar.gz", "msg": "ssh: Could not resolve hostname instance-1: Name or service not known\r\nrsync: connection unexpectedly closed (0 bytes received so far) [Receiver]\nrsync error: unexplained error (code 255) at io.c(235) [Receiver=3.1.3]\n", "rc": 255}
```
The task is like
```
  - name: download_file | Copy file back to ansible host file cache
    synchronize:
      src: "{{ file_path_cached }}"
      dest: "{{ file_path_cached }}"
      use_ssh_args: "{{ has_bastion | default(false) }}"
      mode: pull
    when:
    - download_force_cache
    - not download_localhost
    - download_delegate == inventory_hostname
 ```

Reply:
https://docs.ansible.com/ansible/latest/collections/ansible/posix/synchronize_module.html#parameter-use_ssh_args
```
In Ansible 2.10 and lower, it uses the ssh_args specified in ansible.cfg.
In Ansible 2.11 and onwards, when set to true, it uses all SSH connection configurations like ansible_ssh_args, ansible_ssh_common_args, and ansible_ssh_extra_args.
```

Reply:
After specifying it, that is still failed:
```
fatal: [instance-1]: FAILED! => {"changed": false, "cmd": "/usr/bin/rsync --delay-updates -F --compress --archive --rsh='/usr/bin/ssh -S none -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -C -o ControlMaster=auto -o ControlPersist=60s' --rsync-path='sudo -u root rsync' --out-format='<<CHANGED>>%i %n%L' kubespray@instance-1:/tmp/kubespray_cache/nerdctl-0.19.0-linux-amd64.tar.gz /tmp/kubespray_cache/nerdctl-0.19.0-linux-amd64.tar.gz", "msg": "ssh: Could not resolve hostname instance-1: Name or service not known\r\nrsync: connection unexpectedly closed (0 bytes received so far) [Receiver]\nrsync error: unexplained error (code 255) at io.c(235) [Receiver=3.1.3]\n", "rc": 255}
```
ansible.conf looks like
```
ssh_args = -o ControlMaster=auto -o ControlPersist=30m -o ConnectionAttempts=100 -o UserKnownHostsFile=/dev/null
```
-o ControlMaster=auto: Specified
-o ControlPersist=30m: 60s specified
-o ConnectionAttempts=100: Not specified
-o UserKnownHostsFile=/dev/null: Specified

Then need to know how to specify ControlPersist=30m and ConnectionAttempts=100 on synchronize

Reply:
> I guess the job is missing the entry on /etc/hosts or DNS.

Why would it have worked before if the entry was missing?

Reply:
because  ansible-playbook command knows the target IP(instance-1, etc.) from the inventory file, I guess.